hadoop-common-project/hadoop-annotations/pom.xml

```diff
@@ -75,6 +75,45 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>jdk17</id>
```

Comment:
```suggestion
      <id>jdk17+</id>
```

Reply:
@zhtttylz Could you please test this on JDK 21 to confirm it passes? Thanks a lot.